src/flamegraph/mod.rs

```diff
@@ -325,19 +333,51 @@ impl Rectangle {
 ///
 /// [differential flame graph]: http://www.brendangregg.com/blog/2014-11-09/differential-flame-graphs.html
 #[allow(clippy::cyclomatic_complexity)]
-pub fn from_sorted_lines<'a, I, W>(opt: &mut Options, lines: I, writer: W) -> quick_xml::Result<()>
+pub fn from_lines<'a, I, W>(opt: &mut Options, lines: I, writer: W) -> quick_xml::Result<()>
 where
     I: IntoIterator<Item = &'a str>,
     W: Write,
 {
-    let (bgcolor1, bgcolor2) = color::bgcolor_for(opt.bgcolors, opt.colors);
+    let mut reversed = StrStack::new();
+    let (mut frames, time, ignored, delta_max) = if !opt.no_sort && !opt.reverse_stack_order {
+        // Sort lines by default.
+        let mut lines: Vec<&str> = lines.into_iter().collect();
+        lines.sort_unstable();
+        merge::frames(lines)?
+    } else if opt.reverse_stack_order {
+        if opt.no_sort {
+            warn!("Input lines are always sorted when using --reverse. The --no-sort flag is being ignored.");
+        }
+        // Reverse order of stacks and sort.
+        let mut stack = String::new();
+        for line in lines {
+            stack.clear();
+            let samples_idx = rfind_samples(line).unwrap_or_else(|| line.len());
+            let samples_idx = rfind_samples(&line[..samples_idx - 1]).unwrap_or(samples_idx);
+            for (i, func) in line[..samples_idx].trim().split(';').rev().enumerate() {
+                if i != 0 {
+                    stack.push(';');
+                }
+                stack.push_str(func);
+            }
+            stack.push(' ');
+            stack.push_str(&line[samples_idx..]);
+            reversed.push(&stack);
+        }
+        let mut reversed: Vec<&str> = reversed.iter().collect();
+        reversed.sort_unstable();
+        merge::frames(reversed)?
+    } else {
+        // Lines don't need sorting.
+        merge::frames(lines)?
+    };
 
-    let mut buffer = StrStack::new();
-    let (mut frames, time, ignored, delta_max) = merge::frames(lines)?;
```

Comment:
I'm not sure why these lines had to change?

Reply:
The `buffer` is not being used until we start writing the SVG, so I moved it down to right before that.

The second line was replaced by all of the stuff above.